.github/workflows/unittests.yml

```diff
@@ -0,0 +1,47 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python Test and Lint
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    name: Unittest Anonlink ${{ matrix.python }} ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-latest, windows-latest, ubuntu-18.04, ubuntu-20.04]
+        python: ["3.7", "3.8", "3.9"]
+        
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+      - name: Get full Python version
+        id: full-python-version
+        shell: bash
+        run: echo ::set-output name=version::$(python -c "import sys; print('-'.join(str(v) for v in sys.version_info))")
+      - name: Install dependencies
+        shell: bash
+        run: |
+          python -m pip install -U -r requirements.txt
+      - name: Build and install anonlink
+        shell: bash
+        run: |
+          python -m pip install -e .
+      - name: Test with pytest
+        run: |
+          pytest -q
+      - name: Test with extended size 100k
+        if: ${{ matrix.python == 3.9}}
```

Comment:
now that we have quite a few os-es, should we restrict that further? Do we really need to run a 5 minute test on every platform?